commitserver/commit/hydratorhelper_test.go

```diff
@@ -6,12 +6,82 @@ import (
 	"path"
 	"testing"
 
+	securejoin "github.com/cyphar/filepath-securejoin"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
 	"github.com/argoproj/argo-cd/v2/commitserver/apiclient"
 )
 
+func TestWriteForPaths(t *testing.T) {
+	dir := t.TempDir()
+
+	repoUrl := "https://github.com/example/repo"
+	drySha := "abc123"
+	paths := []*apiclient.PathDetails{
+		{
+			Path: "path1",
+			Manifests: []*apiclient.ManifestDetails{
+				{Manifest: `{"kind":"Pod","apiVersion":"v1"}`},
+			},
+			Commands: []string{"command1", "command2"},
+		},
+		{
+			Path: "path2",
+			Manifests: []*apiclient.ManifestDetails{
+				{Manifest: `{"kind":"Service","apiVersion":"v1"}`},
+			},
+			Commands: []string{"command3"},
+		},
+	}
+
+	err := WriteForPaths(dir, repoUrl, drySha, paths)
+	require.NoError(t, err)
+
+	topMetadataPath := path.Join(dir, "hydrator.metadata")
+	topMetadataBytes, err := os.ReadFile(topMetadataPath)
+	require.NoError(t, err)
+
+	var topMetadata hydratorMetadataFile
+	err = json.Unmarshal(topMetadataBytes, &topMetadata)
+	require.NoError(t, err)
+	assert.Equal(t, hydratorMetadataFile{RepoURL: repoUrl, DrySHA: drySha}, topMetadata)
+
+	for _, p := range paths {
```

Comment:
I worry that this re-implements too much of the function it's calling.

How about asserting some simpler stuff:
1. Does the top-level hydrator.metadata exist and contain the repo URL?
2. Does a directory for each path exist?
3. Does each path contain a hydrator.metadata file and a README.md each containing the repo URL?
4. Does each path contain a manifest.yaml containing the word `Pod`?

That'll make sure the function basically works without being too overly-detailed.